common/kvstore/src/main/java/org/apache/spark/util/kvstore/LevelDBIterator.java

```diff
@@ -189,6 +189,7 @@ public synchronized void close() throws IOException {
     if (!closed) {
       it.close();
       closed = true;
+      next = null;
```

Comment:
Do we want to add a `!closed` check to `skip` as well ?

Reply:
Let me check it later

Reply:
```
KVStoreIterator<CustomType1> iter =
      db.view(CustomType1.class).closeableIterator();
    // iter should be true
    assertTrue(iter.hasNext());
    // close iter
    iter.close();
    assertFalse(iter.skip(0));
    assertFalse(iter.skip(1));
```
I test rocksdb as above:

- `skip(0)` returning false, looks right
- `skip(1)`  or `skip(n)` thrown Exception instead of returning false now:

```
java.lang.AssertionError
	at org.rocksdb.AbstractRocksIterator.isValid(AbstractRocksIterator.java:41)
	at org.apache.spark.util.kvstore.RocksDBIterator.skip(RocksDBIterator.java:164)
	at org.apache.spark.util.kvstore.RocksDBSuite.testSkipAfterIteratorClose(RocksDBSuite.java:376)
```

Reply:
`skip(1)` of `LevelDBIterator` also need fix:
```
java.lang.AssertionError: This object has been deleted
	at org.apache.spark.util.kvstore.LevelDBSuite.testSkipAfterDBClose(LevelDBSuite.java:383)
```